twindb_backup/destination/s3.py

```diff
@@ -281,7 +281,20 @@ def _download_object(s3_client, bucket_name, key, read_fd, write_fd):
 
             with os.fdopen(write_fd, 'wb') as w_pipe:
                 try:
-                    s3_client.download_fileobj(bucket_name, key, w_pipe)
+                    retry_interval = 2
```

Comment:
It would be better to implement it as a decorator. You are using the retry functionality at a number of places.

Have a look at this package: https://pypi.python.org/pypi/retry/
I have had good success with it so far.